src/fluentd/src/fluent-plugin-pgjson/lib/fluent/plugin/out_pgjson.rb

```diff
@@ -151,16 +152,18 @@ def write(chunk)
             kind = record["objectSnapshot"]["kind"]
             log.debug "log type: #{kind}"
             if kind == "Framework"
-              thread[:conn].exec("COPY framework_history (#{@insertedAt_col}, #{@frameworkName_col}, #{@attemptIndex_col}, #{@historyType_col}, #{@snapshot_col}) FROM STDIN WITH DELIMITER E'\\x01'")
+              thread[:conn].exec("COPY framework_history (\"#{@insertedAt_col}\", \"#{@updatedAt_col}\", \"#{@uid_col}\", \"#{@frameworkName_col}\", \"#{@attemptIndex_col}\", \"#{@historyType_col}\", \"#{@snapshot_col}\") FROM STDIN WITH DELIMITER E'\\x01'")
               frameworkName = record["objectSnapshot"]["metadata"]["name"]
               attemptIndex = record["objectSnapshot"]["status"]["attemptStatus"]["id"]
               historyType = "retry"
               snapshot = record_value(record["objectSnapshot"])
-              thread[:conn].put_copy_data "#{time}\x01#{frameworkName}\x01#{attemptIndex}\x01#{historyType}\x01#{snapshot}\n"
+              # use frameworkName + attemptIndex + historyType to generate a uid
+              uid = Digest::MD5.hexdigest "#{frameworkName}+#{attemptIndex}+#{historyType}"
+              thread[:conn].put_copy_data "#{time}\x01#{time}\x01#{uid}\x01#{frameworkName}\x01#{attemptIndex}\x01#{historyType}\x01#{snapshot}\n"
             elsif kind == "Pod"
```

Comment:
Add comment that if duplicate, what is the behaiour here?
Crash, retry, ignore or log

Reply:
It will raise an error and log the error.